File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        res, q = 0, deque([root])
        while q:
            l = len(q)
            for _ in range(0, l):
                root = q.popleft()
                if root.left:
                    q.append(root.left)
                if root.right:
                    q.append(root.right)
            res += 1
        return res
    
        #
        
        if not root:
            return 0
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function maxDepth(root: TreeNode | null): number {
    if (!root) return 0
    const q: TreeNode[] = [root]
    let depth = 0
    while (q.length) {
        const len = q.length
        for (let i=0; i < len; i++) {
            const {left, right} = q.shift()
            if (left) q.push(left)
            if (right) q.push(right)
        }
        depth++
    }
    return depth
    
    //
    
    if (!root) return 0
    return 1 + Math.max(maxDepth(root.left), maxDepth(root.right))
};